File: matching-learning-api/matching-learning.common/Domain/DTOs/Evaluation.cs
using System;
using System.Collections.Generic;

namespace matching_learning.common.Domain.DTOs
{
    public class Evaluation : IComparable<Evaluation>
    {
        public int Id { get; set; }

        public int CandidateId { get; set; }

        public string EvaluationKey { get; set; }

        public int EvaluationTypeId { get; set; }

        public EvaluationType EvaluationType { get; set; }

        public DateTime Date { get; set; }
        
        public string Notes { get; set; }

        public List<EvaluationDetail> Details { get; set; }

        public int CompareTo(Evaluation other)
        {
            return (-1 * Date.CompareTo(other.Date));
        }
    }
}

File: matching-learning/src/app/shared/models/evaluation.ts
import { EvaluationDetails } from './evaluation-details';
import {EvaluationType} from './evaluation-type';
import {Skill} from './skill';


export interface Evaluation {
   id: number;
   candidateId: number;
   evaluationType: EvaluationType;
   evaluationTypeId: number;
   date: Date;
   details: EvaluationDetails[];
   notes: string;
}
